
# Computational & Statistical Systems Biology Laboratory

CSSB Laboratory develops computational and statistical solutions for the analysis of high-dimensional molecular data in various contexts of biomedical research. These tools are motivated by a wide range of research topics, including

- Integration of multiple molecular data sets to harness systems-level biological information
- Statistical methods for ultra high-dimensional data
- Computational tool development for mass spectrometry (MS) data
- Interactive data visualization

CSSB Laboratory also provides massively parallel sequencing (or next generation sequencing (NGS)) and MS data analysis support for basic scientists and clinical investigators through Systems Biology Core at NUHS. Our collaborative service work includes exome sequencing, epigenetic profiling, transcriptomics, proteomics and metabolomics/lipidomics data analysis.

# Check out our latest softwares in GitHub and SourceForge repositories!

- iOmicsPASS+ (https://github.com/cssblab/iOmicsPASSplus): Extension of iOmicsPASS with direct partial correlation network estimation for general multi-omics data

- GPD (https://github.com/ginnyintifa/GPD): annotation-based mapping protein coding variants to structural units of protein sequences for downstream statistical analysis of cancer genomics data (<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/humu.23979" paper </a>)

- ParProx (https://github.com/kose-y/ParProx.jl): parallel or distributed computing for ultrahigh-dimensional group lasso regression (paper - https://academic.oup.com/bib/article/22/6/bbab256/6319937)

- KSA2D (https://github.com/ginnyintifa/KSA2D): sensitive differential PTM analysis and kinase-substrate joint differential analysis

- eGFR smoothing (https://github.com/jleechung/eGFRsmooth): clinical time course data smoothing and detection of rapid decline in functional parameters (e.g. glomerular filtration rates for kidney disease)

- MRMkit (https://github.com/MRMkit/MRMkit): Automated peak integration and data normalization for targeted mass spectrometry in large-scale studies (paper - https://pubs.acs.org/doi/10.1021/acs.analchem.0c03060)

- MetaboKit (https://github.com/MetaboKit/MetaboKit): Data dependent and independent acquisition mass spectrometry analyiss workflow for untargeted metabolomics experiments (paper - https://pubs.rsc.org/en/content/articlelanding/2020/mo/d0mo00030b)

- iOmicsPASS (https://github.com/cssblab/iOmicsPASS): Integrative -omics approach for predictive analysis of cancer subtypes with sub-network signatures (paper - https://www.nature.com/articles/s41540-019-0099-y) 

- PTMscape (http://github.com/ginnyintifa/PTMscape/): an open source tool to predict generic post-translational modifications and map hotspots of modification crosstalk (paper - https://pubs.rsc.org/en/content/articlelanding/2018/mo/c8mo00027a)

- multiSLIDE (https://github.com/soumitag/multiSLIDE): heatmap visualization of selected molecular features of quantitative multi-omics data (paper - https://www.nature.com/articles/s41467-021-22650-x)

- EBprot (http://github.com/cssblab/EBprot): Statistical analysis of labeling-based proteomics data. First version in R package is available at http://sourceforge.net/projects/ebprot/ (paper - https://pubs.acs.org/doi/10.1021/acs.jproteome.8b00483)

- PECA (https://github.com/PECAplus): Protein expression control analysis for gene expression regulation via multi-omics data integration in time series (paper - https://pubs.acs.org/doi/abs/10.1021/pr400855q)

- mapDIA (http://github.com/mapDIA/mapDIA): Data processing and statistical analysis of data independent acquisition MS data (in fact, any -omics data!) (paper - https://www.sciencedirect.com/science/article/abs/pii/S1874391915301305?via%3Dihub)

- QSPEC/QPROT (http://sourceforge.net/projects/qprot/): Statistical protein differential expression analysis for spectral count data (paper - https://www.sciencedirect.com/science/article/abs/pii/S1874391915300907)

- LuciPHOr (http://luciphor.sourceforge.net/, http://luciphor2.sourceforge.net/): Probabilistic algorithm to control false localization rates of generic PTM from mass spectrometry-based proteomics data (paper - https://www.mcponline.org/article/S1535-9476(20)34616-8/fulltext)

# Upcoming Software

- Next generation MS-based metabolomics tools: high-resolution MS acquisition with and without chromatographic separation

- Ultrahigh-dimensional sparse precision matrix estimation with and without limit of detection in the data via distributed computing

- Structure-assisted chemoinformatic solutions for compound discovery and identification
